osu.Game/Rulesets/Mods/ModBarrelRoll.cs

```diff
@@ -50,8 +50,8 @@ public void ApplyToDrawableRuleset(DrawableRuleset<TObject> drawableRuleset)
 
             var playfieldSize = drawableRuleset.Playfield.DrawSize;
             var minSide = MathF.Min(playfieldSize.X, playfieldSize.Y);
-            var maxSide = MathF.Max(playfieldSize.X, playfieldSize.Y);
-            drawableRuleset.Playfield.Scale = new Vector2(minSide / maxSide);
+            var diagonal = MathF.Sqrt(MathF.Pow(playfieldSize.X, 2) + MathF.Pow(playfieldSize.Y, 2));
+            drawableRuleset.Playfield.Scale = new Vector2(minSide / diagonal);
```

Comment:
This is what `playfieldSize.Length` would do, which even has a faster version (`playfieldSize.LengthFast`) that is already used in `OsuModRandom`.